### ISSUE TITLE
500 error if you remove a file belonging to an album

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/photos#1226]
- [ ] Watch for file deletion event and edit the album accordingly

```json
{
  "reqId": "ozZ8J8lLHpAMUxHvqoaX",
  "level": 3,
  "time": "2022-09-08T21:22:42+00:00",
  "remoteAddr": "172.18.0.1",
  "user": "admin",
  "app": "webdav",
  "method": "PROPFIND",
  "url": "/remote.php/dav/photos/admin/albums/",
  "message": "Photo not found for user",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36",
  "version": "25.0.0.8",
  "exception": {
    "Exception": "OCP\\Files\\NotFoundException",
    "Message": "Photo not found for user",
    "Code": 0,
    "Trace": [
      {
        "file": "/var/www/nextcloud/apps2/photos/lib/Sabre/Album/AlbumRoot.php",
        "line": 164,
        "function": "getFileInfo",
        "class": "OCA\\Photos\\Sabre\\Album\\AlbumPhoto",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/nextcloud/apps2/photos/lib/Sabre/Album/PropFindPlugin.php",
        "line": 111,
        "function": "getDateRange",
        "class": "OCA\\Photos\\Sabre\\Album\\AlbumRoot",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/PropFind.php",
        "line": 95,
        "function": "OCA\\Photos\\Sabre\\Album\\{closure}",
        "class": "OCA\\Photos\\Sabre\\Album\\PropFindPlugin",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/var/www/nextcloud/apps2/photos/lib/Sabre/Album/PropFindPlugin.php",
        "line": 111,
        "function": "handle",
        "class": "Sabre\\DAV\\PropFind",
        "type": "->",
        "args": [
          "{http://nextcloud.org/ns}dateRange",
          {
            "__class__": "Closure"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "propFind",
        "class": "OCA\\Photos\\Sabre\\Album\\PropFindPlugin",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\DAV\\PropFind"
          },
          {
            "__class__": "OCA\\Photos\\Sabre\\Album\\AlbumRoot",
            "rootFolder": {
              "__class__": "OC\\Files\\Node\\LazyRoot"
            }
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1052,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          "propFind",
          [
            {
              "__class__": "Sabre\\DAV\\PropFind"
            },
            {
              "__class__": "OCA\\Photos\\Sabre\\Album\\AlbumRoot",
              "rootFolder": {
                "__class__": "OC\\Files\\Node\\LazyRoot"
              }
            }
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 984,
        "function": "getPropertiesByNode",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\DAV\\PropFind"
          },
          {
            "__class__": "OCA\\Photos\\Sabre\\Album\\AlbumRoot",
            "rootFolder": {
              "__class__": "OC\\Files\\Node\\LazyRoot"
            }
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1662,
        "function": "getPropertiesIteratorForPath",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          "photos/admin/albums",
          [
            "{http://nextcloud.org/ns}last-photo",
            "{http://nextcloud.org/ns}nbItems",
            "{http://nextcloud.org/ns}location",
            "{http://nextcloud.org/ns}dateRange"
          ],
          1
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 1647,
        "function": "writeMultiStatus",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\Xml\\Writer",
            "elementMap": [],
            "contextUri": "/remote.php/dav/",
            "namespaceMap": {
              "DAV:": "d",
              "http://sabredav.org/ns": "s",
              "http://owncloud.org/ns": "oc",
              "http://nextcloud.org/ns": "nc"
            },
            "classMap": []
          },
          {
            "__class__": "Generator"
          },
          false
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php",
        "line": 346,
        "function": "generateMultiStatus",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          {
            "__class__": "Generator"
          },
          false
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php",
        "line": 89,
        "function": "httpPropFind",
        "class": "Sabre\\DAV\\CorePlugin",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\HTTP\\Request"
          },
          {
            "__class__": "Sabre\\HTTP\\Response"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 472,
        "function": "emit",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          "method:PROPFIND",
          [
            {
              "__class__": "Sabre\\HTTP\\Request"
            },
            {
              "__class__": "Sabre\\HTTP\\Response"
            }
          ]
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 253,
        "function": "invokeMethod",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": [
          {
            "__class__": "Sabre\\HTTP\\Request"
          },
          {
            "__class__": "Sabre\\HTTP\\Response"
          }
        ]
      },
      {
        "file": "/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php",
        "line": 321,
        "function": "start",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/nextcloud/apps/dav/lib/Server.php",
        "line": 360,
        "function": "exec",
        "class": "Sabre\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/nextcloud/apps/dav/appinfo/v2/remote.php",
        "line": 35,
        "function": "exec",
        "class": "OCA\\DAV\\Server",
        "type": "->",
        "args": []
      },
      {
        "file": "/var/www/nextcloud/remote.php",
        "line": 167,
        "args": [
          "/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "/var/www/nextcloud/apps2/photos/lib/Sabre/Album/AlbumPhoto.php",
    "Line": 102,
    "message": "Photo not found for user",
    "exception": {},
    "CustomMessage": "Photo not found for user"
  }
}
```

**Comment by artonge** [comment]
Seen and is fixed in https://github.com/nextcloud/photos/pull/1161 ;)

**Comment by skjnldsv** [comment]
> Seen and is fixed in #1161 ;)

Maybe better split off :)
